homeassistant/components/gios/air_quality.py

```diff
@@ -0,0 +1,174 @@
+"""Support for the GIOS service."""
+from homeassistant.components.air_quality import (
+    ATTR_CO,
+    ATTR_NO2,
+    ATTR_OZONE,
+    ATTR_PM_2_5,
+    ATTR_PM_10,
+    ATTR_SO2,
+    AirQualityEntity,
+)
+from homeassistant.const import CONF_NAME
+
+from .const import ATTR_STATION, DATA_CLIENT, DOMAIN
+
+ATTRIBUTION = "Data provided by GIOŚ"
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add a GIOS entities from a config_entry."""
+    name = config_entry.data[CONF_NAME]
+
+    data = hass.data[DOMAIN][DATA_CLIENT][config_entry.entry_id]
+
+    async_add_entities([GiosAirQuality(data, name)], True)
+
+
+def round_state(func):
+    """Round state."""
+
+    def _decorator(self):
+        res = func(self)
+        if isinstance(res, float):
+            return round(res)
+        return res
+
+    return _decorator
+
+
+class GiosAirQuality(AirQualityEntity):
+    """Define an GIOS sensor."""
+
+    def __init__(self, gios, name):
+        """Initialize."""
+        self.gios = gios
+        self._name = name
+        self._aqi = None
+        self._co = None
+        self._no2 = None
+        self._o3 = None
+        self._pm_2_5 = None
+        self._pm_10 = None
+        self._so2 = None
+        self._icon = None
+        self._attrs = {}
+
+    @property
+    def name(self):
+        """Return the name."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        if self._aqi == "bardzo dobry":
+            self._icon = "mdi:emoticon-excited"
+        elif self._aqi == "dobry":
+            self._icon = "mdi:emoticon-happy"
+        elif self._aqi == "umiarkowany":
+            self._icon = "mdi:emoticon-neutral"
+        elif self._aqi == "dostateczny":
+            self._icon = "mdi:emoticon-sad"
+        elif self._aqi == "zły":
+            self._icon = "mdi:emoticon-dead"
+        else:
+            self._icon = "mdi:blur"
+        return self._icon
+
+    @property
+    def air_quality_index(self):
+        """Return the air quality index."""
+        return self._aqi
+
+    @property
+    @round_state
```

Comment:
Should rounding be left to the consumers of the data? I don't personally mind, but I think someone might :)

Reply:
@rytilahti In Poland, most air quality services on the Internet rounding off values. I think this is a good approach.